fastparquet/api.py

```diff
@@ -371,6 +376,15 @@ def remove_row_groups(self, rgs, write_fmd:bool = True,
             (and any contained files). Not required if this ParquetFile has
             a .fs file system attribute
         """
+        if not isinstance(rgs, list):
+            if not hasattr(rgs, '__iter__'):
```

Comment:
Thanks, I rewrote the code.
I did not keep the `try/except` though, because lines of code immediately after take care of empty row-groups list.
This `isinstance/hasattr` has been here to catch row group list given as generator.

I left a comment explaining why I swith from `[]` (previous code) to `list()`.
- On a generator, `[]` encapsulate it in a list. When you are iterating over the list content, you get the generator, not the items of the generator.
- `list()` materializes the generator into a list. when you iterate over it, it will iterate the row groups.

Reply:
Does the first branch ever get called? A tuple also has `__iter__`, actually many things do. If what you really want to check for generator, why not do exactly that (`inspect.isgenerator`) ?